modules/web/js/ballerina/components/transform-render.js

```diff
@@ -35,858 +35,891 @@ var jsPlumb = jsPlumbLib.jsPlumb;
+}
 
-    /**
-     * Populate the attribute properties stack from the propertyId
-     * @param propertyId
-     * @returns {Array}
-     */
-    getPropertyNameStack(propertyId) {
-        var id = propertyId.replace(this.jsTreePrefix + this.viewIdSeperator, "");
-        var parts = id.split(this.idNameSeperator);
-        var propertyNames = [];
-        var self = this;
-        var elementId = 0;
-        _.forEach(parts, aPart => {
-            if (elementId != 0) {
-                propertyNames.push(aPart.split(self.nameTypeSeperator)[0]);
-            } else {
-                elementId++;
-            }
-        });
-        return propertyNames;
-    }
+/**
+ * Get the name of the property from the property id
+ * @param propertyId
+ * @returns {*}
+ */
+getPropertyName(propertyId) {
+    var parts = propertyId.split(this.idNameSeperator);
+    return parts[parts.length - 1].split(this.nameTypeSeperator)[0];
+}
 
-    /**
-     * Process the connection and determines whether to show/hide the dropdown from the connection.
-     * @param info connection object.
-     */
-    processTypeMapperDropdown(info) {
-        var sourceType = this.getPropertyType(info.sourceId);
-        var targetType = this.getPropertyType(info.targetId);
-        var isValidTypes = sourceType == targetType;
-        var self = this;
-        if (!isValidTypes) {
-            var connection = info.connection;
-            connection.getOverlay("typeMapperDropdown").show();
-            var typeMapperId = '#typeMapperList' + this.viewIdSeperator + this.viewId;
-            var updatedTypeMapperId = "typeMapperList" + this.viewIdSeperator + connection.sourceId
+/**
+ * Populate the attribute properties stack from the propertyId
+ * @param propertyId
+ * @returns {Array}
+ */
+getPropertyNameStack(propertyId) {
+    var id = propertyId.replace(this.jsTreePrefix + this.viewIdSeperator, "");
+    var parts = id.split(this.idNameSeperator);
+    var propertyNames = [];
+    var self = this;
+    var elementId = 0;
+    _.forEach(parts, aPart => {
+        if (elementId != 0) {
+        propertyNames.push(aPart.split(self.nameTypeSeperator)[0])
+    } else {
+        elementId++;
+    }
+});
+return propertyNames;
+}
+
+/**
+ * Process the connection and determines whether to show/hide the dropdown from the connection.
+ * @param info connection object.
+ */
+processTypeMapperDropdown(info) {
+    var sourceType = this.getPropertyType(info.sourceId);
+    var targetType = this.getPropertyType(info.targetId);
+    var isValidTypes = sourceType == targetType;
+    var self = this;
+    if (!isValidTypes) {
+        var connection = info.connection;
+        connection.getOverlay("typeMapperDropdown").show();
+        var typeMapperId = '#typeMapperList' + this.viewIdSeperator + this.viewId;
+        var updatedTypeMapperId = "typeMapperList" + this.viewIdSeperator + connection.sourceId
```

Comment:
We need to get rid of double quotes as per our eslint rules. I think theres a command to fix them automatically, @roonyh would know the command. We just need to give the path as a param.